elasticapm/handlers/logging.py

```diff
@@ -155,3 +156,55 @@ def _emit(self, record, **kwargs):
             logger_name=record.name,
             **kwargs
         )
+
+
+class LoggingFilter(logging.Filter):
+    """
+    This filter doesn't actually do any "filtering" -- rather, it just adds
+    two new attributes to any "filtered" LogRecord objects:
+
+    * elasticapm_transaction_id
+    * elasticapm_trace_id
+    * elasticapm_span_id
+
+    These attributes can then be incorporated into your handlers and formatters,
+    so that you can tie log messages to transactions in elasticsearch.
+    """
+
+    def filter(self, record):
+        """
+        Add elasticapm attributes to `record`.
+        """
+        transaction = execution_context.get_transaction()
+        record.elasticapm_transaction_id = transaction.id if transaction else None
+        record.elasticapm_trace_id = (
+            transaction.trace_parent.trace_id if transaction and transaction.trace_parent else None
+        )
+        span = execution_context.get_span()
+        record.elasticapm_span_id = span.id if span else None
+        return True
+
+
+def structlog_processor(logger, method_name, event_dict):
```

Comment:
I wonder if we should create a `elasticapm.contrib.structlog` module and move the processor there, since this isn't really related to the `logging` stdlib module.

Reply:
Done.